src/core/components/hv-navigator/index.tsx

```diff
@@ -6,43 +6,264 @@
+    element: TypesLegacy.Element,
+    type: TypesLegacy.DOMString,
+  ): React.ReactNode => {
+    const screens: React.ReactElement[] = [];
+    const navProps: NavigationContext.NavigationContextProps | null = useContext(
+      NavigationContext.Context,
+    );
+    const navCache: NavigatorContext.NavigatorCache | null = useContext(
+      NavigatorContext.NavigatorMapContext,
+    );
+    if (!navProps || !navCache) {
+      throw new NavigatorService.HvRouteError('No context found');
     }
-    if (!this.state.doc) {
-      const loadingScreen = this.context.loadingScreen || Loading;
-      return React.createElement(loadingScreen);
+
+    const elements: TypesLegacy.Element[] = NavigatorService.getChildElements(
+      element,
+    );
+    const { buildTabScreen } = this;
+    for (let i = 0; i < elements.length; i += 1) {
+      const child: TypesLegacy.Element = elements[i];
+      if (child.localName === TypesLegacy.LOCAL_NAME.NAV_ROUTE) {
+        const id: TypesLegacy.DOMString | null | undefined = child.getAttribute(
+          'id',
+        );
+        if (!id) {
+          throw new NavigatorService.HvNavigatorError(
+            `No id provided for ${child.localName}`,
+          );
+        }
+
+        // Check for nested navigators
+        const nestedNavigator: TypesLegacy.Element | null = getFirstTag(
+          child,
+          TypesLegacy.LOCAL_NAME.NAVIGATOR,
+        );
+        if (nestedNavigator) {
+          // Cache the navigator for the route
+          navCache.elementMap?.set(id, nestedNavigator);
+        } else {
+          const href:
+            | TypesLegacy.DOMString
+            | null
+            | undefined = child.getAttribute('href');
+          if (!href) {
+            throw new NavigatorService.HvNavigatorError(
+              `No href provided for route '${id}'`,
+            );
+          }
+          const url = NavigatorService.getUrlFromHref(
+            href,
+            navProps?.entrypointUrl,
+          );
+
+          // Cache the url for the route
+          navCache.routeMap?.set(id, url);
+        }
+
+        // Stack uses route urls, other types build out the screens
+        if (type !== NavigatorService.NAVIGATOR_TYPE.STACK) {
+          screens.push(buildTabScreen(id, type));
+        }
+      }
     }
 
+    // Add the dynamic screens
+    switch (type) {
+      case NavigatorService.NAVIGATOR_TYPE.STACK:
+        // Dynamic is used to display all routes in stack which are presented as cards
+        screens.push(
+          <Stack.Screen
+            key={NavigatorService.ID_DYNAMIC}
+            component={HvRoute}
+            getId={({ params }) => params?.url}
+            // empty object required because hv-screen doesn't check for undefined param
+            initialParams={{}}
+            name={NavigatorService.ID_DYNAMIC}
+          />,
+        );
+
+        // Modal is used to display all routes in stack which are presented as modals
+        screens.push(
```

Comment:
are we missing a `case` statement here?

Reply:
No missing case statement (now if clause in updated code). The two screens (dynamic and modal) are added to every navigator of type STACK